sql/flow_test.go

```diff
@@ -54,9 +69,42 @@ func TestCommonDockerUtilSuccess(t *testing.T) {
 		mockDockerBinder.On("ContainerLogs", mock.Anything, mock.Anything, mock.Anything).Return(sampleLog, nil)
 		return mockDockerBinder, nil
 	}
+	PrintBuildingSteps = mockPrintBuildingStepsNil
 	err := CommonDockerUtil(testCommand, nil, map[string]string{"flag": "value"}, []string{"mountDirectory"})
 	assert.NoError(t, err)
 	mockDockerBinder.AssertExpectations(t)
+	PrintBuildingSteps = printBuildingSteps
+}
+
+func TestPrintBuildingSteps(t *testing.T) {
+	expectedOutput := `Installing flow.. This might take some time.
+Step 2/4 : ENV ASTRO_CLI Yes
+`
+	// Capture output from PrintBuildingSteps Println calls
+	actualOutput := ""
+	Println = func(a ...any) (n int, err error) {
+		for _, arg := range a {
+			actualOutput += fmt.Sprintln(arg)
+		}
+		return 0, nil
+	}
```

Comment:
a simpler way to do the same thing would be to listen to `os.Stdout` (Reference: https://github.com/astronomer/astro-cli/blob/main/airflow/docker_test.go#L673-L691) and capture everything. IMO ideally, monkey-patching the code to write unit tests should be the last resort, it makes code less readable and prone to misuse.

Reply:
That's exactly what I (and @pankajkoti) were looking for. Thanks.